### ISSUE TITLE
U_FILE_ACCESS_ERROR with tensorflow_text

### ISSUE THREAD

**Issue by tgsmith61591** [tensorflow/serving#1613]
### System information
```
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial
```
- **TensorFlow Serving installed from (source or binary)**: Using a stock docker image
- **TensorFlow Serving version**: `tensorflow/serving:nightly`
- **TensorFlow version**: `2.2.0rc3`
- **TensorFlow Text version**: `2.2.0.rc2`
- **Python version**: `3.6.10`

### Describe the problem

This was posted as an issue on the tensorflow/text repo (https://github.com/tensorflow/text/issues/281), but I suspect since it has to do with the way serving is built, I think it belongs here too. When serving a model that has a `BERTTokenizer` from `tf_text`, I'm getting:

```
{'error': 'U_FILE_ACCESS_ERROR: Could not retrieve ICU NFKC_CaseFold normalizer\\n\\t [[{{node transformer_network/bert_embeddings/bert-tokenizer/CaseFoldUTF8/CaseFoldUTF8}}]]'}
```

I suspect `tensorflow_text` is not successfully built as part of the `third_party` dependencies in tensorflow_serving, which is why it cannot find the op.

**Comment by tgsmith61591** [comment]
For whatever it's worth, I built `tensorflow_serving` with the following changes:

1. Updated the `workspace.bzl` to point to the new tf-text distro:

  ```bazel
      http_archive(
          name = "org_tensorflow_text",
          sha256 = "468abebaa91b08fc0bd34ca7b8862ff7d7853556f231ba9b06f887c3aaf5f1bc",
          strip_prefix = "text-2.2.0-rc2",
          urls = [
              "https://github.com/tensorflow/text/archive/v2.2.0-rc2.zip",
          ],
          patches = ["@//third_party/tf_text:tftext.patch"],
          patch_args = ["-p1"],
          repo_mapping = {"@com_google_re2": "@com_googlesource_code_re2"},
      )
  ```

2. Updated the `tftext.patch` to:

  ```diff
diff -Naur a/tensorflow_text/core/kernels/BUILD b/tensorflow_text/core/kernels/BUILD
--- a/tensorflow_text/core/kernels/BUILD	2020-04-27 14:23:30.000000000 -0500
+++ b/tensorflow_text/core/kernels/BUILD	2020-04-27 15:39:35.000000000 -0500
@@ -16,8 +16,7 @@
     "@com_google_absl//absl/strings",
     "@com_google_absl//absl/types:optional",
     "@com_google_absl//absl/types:span",
-    "@local_config_tf//:libtensorflow_framework",
-    "@local_config_tf//:tf_header_lib",
+    "@org_tensorflow//tensorflow/core:tensorflow_opensource",
 ]
 
 cc_library(
diff -Naur a/tensorflow_text/tftext.bzl b/tensorflow_text/tftext.bzl
--- a/tensorflow_text/tftext.bzl	2020-04-27 14:23:30.000000000 -0500
+++ b/tensorflow_text/tftext.bzl	2020-04-27 15:41:08.000000000 -0500
@@ -44,8 +44,7 @@
         copts = [ "-pthread", ],
         alwayslink = 1,
         deps = cc_op_kernels + [
-            "@local_config_tf//:libtensorflow_framework",
-            "@local_config_tf//:tf_header_lib",
+            "@org_tensorflow//tensorflow/core:tensorflow_opensource",
         ],
     )
 
@@ -55,8 +54,7 @@
         linkshared = 1,
         deps = [
             ":" + library_name,
-            "@local_config_tf//:libtensorflow_framework",
-            "@local_config_tf//:tf_header_lib",
+            "@org_tensorflow//tensorflow/core:tensorflow_opensource",
         ],
     )
  ```

The build was successful, but I'm getting the same error when I serve the model:
```
{ "error": "U_FILE_ACCESS_ERROR: Could not retrieve ICU NFKC_CaseFold normalizer\n\t [[{{node transformer_network/bert_embeddings/bert-tokenizer/CaseFoldUTF8/CaseFoldUTF8}}]]" }
```

Maybe that helps someone... Tensorflow-text has an ICU patch that I notice is not included in serving. Perhaps that is something to look into?

**Comment by tgsmith61591** [comment]
@lilao bump. I have a PR for this but no one has acknowledged it.

**Comment by lilao** [comment]
@broken replied to your pull request (https://github.com/tensorflow/serving/pull/1616). Please let us know if you have any questions.

**Comment by broken** [comment]
Most of this is info is on the corresponding PR, but adding here as well.

The problem was missing ICU data. All of ICU data was supposed to be built into TF model server, but was no longer (either from a Bazel update or likely it wasn't to begin with). I talked with the ICU team on how to do this and it required a different patch that inserted Bazel build files in different directories of the ICU library, building tools that converted the data, and building the data which is different depending on the platform. While I close to working, ultimately this became way too complicated and fragile. Thus, we shifted to combining the patch files of TF and TF Text builds here.

This was initially not caught because the regression test made use of tf.control_dependencies which was swallowing the errors. TF core is now aware of this, and we are working around it through the use of tf.Assert.

Separately, the icu archive from github has changed (thus changing the validation sha), so we will be using a Google mirror to ensure its stability. (This change will be in soon.)
